src/main/java/dtb/lego/truck/rest/data/events/acquisition/control/DataStreamingHandler.java

```diff
@@ -0,0 +1,140 @@
+     * since it needs to be checked by the EventListeners and the streamers which run in separate threads.
+     */
+    private ConcurrentMap<String, String> subscriptions;
+    @Autowired
+    private SensorDataHandler sensorDataHandler;
+
+    @Autowired
+    public DataStreamingHandler(MessageSendingOperations<String> messagingTemplate) {
+        this.messagingTemplate = messagingTemplate;
+        this.subscriptions = new ConcurrentHashMap<>();
+    }
+
+
+    /**
+     * Event listener that runs whenever a new client subscribes to a topic. It extracts the information regarding the stream
+     * in order to call {@link #streamer(String, String, int, String, String) streamer} function in the requested way, as
+     * well as add the session ID and destination to the current subscriptions map
+     *
+     * @param event Contains the information regarding the subscription
+     */
+    @Async
+    @EventListener
+    public void onSubscribeApplicationEvent(SessionSubscribeEvent event) {
+        SimpMessageHeaderAccessor headers = SimpMessageHeaderAccessor.wrap(event.getMessage());
+        String destination = headers.getDestination();
+        String[] subtopics = destination.split("/");
+        Integer interval = Integer.parseInt(subtopics[2]);
+        String sessionID = headers.getSessionId();
+        String fields = subtopics[3];
+        String transformation = subtopics[4];
+        subscriptions.putIfAbsent(sessionID, destination);
+        streamer(headers.getSessionId(), destination, interval, fields, transformation);
+    }
+
+    /**
+     * Handles the streaming of data to the client. It loops until the client unsubscribe and its sessionId is removed from
+     * the streaming list.
+     *
+     * @param sessionId      Client sessionID
+     * @param destination    The topic to be published
+     * @param interval       The interval in which the data is sent
+     * @param fields         The components from which the data is requested, separated by -
+     * @param transformation The transformation to be applied to downsample the data.
+     */
+    @Async("AsyncWebsocketExecutor")
+    public void streamer(String sessionId, String destination, int interval, String fields, String transformation) {
+        // Get different components from request
+        String[] components = fields.split("-");
+        while (true) {
+            // If the client disconnects his sessionId is removed, so the loop should stop
+            if (!this.subscriptions.containsKey(sessionId)) break;
+            // Send the last data from each component
+            try {
+                for (String component : components) {
+                    Event ne = sensorDataHandler.getTransformedEvent(interval, component, transformation);
+                    messagingTemplate.convertAndSend(destination, ne);
+                }
+                Thread.sleep(interval);
+            } catch (NullPointerException e) {
```

Comment:
It's preferable to check for null than it is to catch NullPointerException.

Reply:
Fixed, thanks